### PR TITLE
feat: Added the ability to use the TEACH-ME mode

### PR DESCRIPTION
**Pull request by FroVolod**
```
./cargo-near --teach-me near new test
```
Screenshot 2024-10-03 at 19 51 28

```
./cargo-near --teach-me near build --manifest-path /Users/frovolod/Documents/NEAR/neardevhub-contract-1/Cargo.toml
```
Screenshot 2024-10-04 at 08 08 49

```
./cargo-near --teach-me near deploy --manifest-path /Users/frovolod/Documents/NEAR/neardevhub-contract-1/Cargo.toml
```

https://github.com/user-attachments/assets/9c205faa-4e0f-4b13-a4c8-66e408ed1346

### CODE REVIEW

**Comment by FroVolod** [comment]
@race-of-sloths

**Comment by FroVolod** [comment]
Three alternative commands were tested:
```
./cargo-near --teach-me near abi --manifest-path /Users/frovolod/Documents/NEAR/neardevhub-contract-1/Cargo.toml
```
```
./cargo-near near --teach-me abi --manifest-path /Users/frovolod/Documents/NEAR/neardevhub-contract-1/Cargo.toml
```
```
cargo near --teach-me abi --manifest-path /Users/frovolod/Documents/NEAR/neardevhub-contract-1/Cargo.toml
```

The execution of the program is the same:
Screenshot 2024-10-11 at 15 54 25

File: cargo-near/src/commands/new/mod.rs
Comment (dj8yfo) [inline cargo-near/src/commands/new/mod.rs:149]: lost during merge operation 
image

Comment (FroVolod) [reply cargo-near/src/commands/new/mod.rs:141]: done

File: cargo-near/src/commands/new/mod.rs
Comment (dj8yfo) [inline cargo-near/src/commands/new/mod.rs:98]: This snippet also landed on main https://github.com/near/cargo-near/commit/fa934ed4a5cf1f985a0ba6b8c296566690b9de56

Comment (FroVolod) [reply cargo-near/src/commands/new/mod.rs:101]: done

**Comment by FroVolod** [comment]
https://github.com/user-attachments/assets/547f349e-5447-4de5-b859-03b92d14fbf3

File: cargo-near/src/commands/new/mod.rs
Comment (dj8yfo) [inline cargo-near/src/commands/new/mod.rs:100]: this one doesn't get displayed  on `--teach-me`, but we probably can figure this out in subsequent prs

File: cargo-near-build/src/env_keys.rs
Comment (dj8yfo) [inline cargo-near-build/src/env_keys.rs:47]: rust's debug representation `"{:#?}"` should be probably used as is here (and wrap in `'...'` single quotes on line 43 removed too)

**Review by dj8yfo — DISMISSED** [review]

**Comment by dj8yfo** [comment]
@race-of-sloths score 13

**Comment by race-of-sloths** [comment]
@FroVolod Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!


[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/FroVolod?type=bot&pr=near/cargo-near/221&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/FroVolod?type=bot&pr=near/cargo-near/221&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/FroVolod)


<details><summary>Current status: <i>waiting for merge</i></summary>

| Reviewer | Score |
|--------|--------|
| @dj8yfo  | 13 |




Your contribution is much appreciated with a final score of 13!
You have received 130 Sloth points for this contribution

@dj8yfo received 25 Sloth Points for reviewing and scoring this pull request.
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/FroVolod)

For maintainers:
- Score pull requests that participate in the Race of Sloths and receive a reward
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: cargo-near-build/src/types/cargo/metadata.rs
Comment (FroVolod) [reply cargo-near-build/src/types/cargo/metadata.rs:162]: Done

File: cargo-near/src/lib.rs
Comment (FroVolod) [reply cargo-near/src/lib.rs:17]: Done

File: cargo-near-build/src/cargo_native/compile.rs
Comment (FroVolod) [reply cargo-near-build/src/cargo_native/compile.rs]: Done

Comment (FroVolod) [reply cargo-near-build/src/cargo_native/compile.rs]: Done

File: cargo-near-build/src/near/docker_build/subprocess_step.rs
Comment (FroVolod) [reply cargo-near-build/src/near/docker_build/subprocess_step.rs]: Done

File: cargo-near-build/src/near/docker_build/subprocess_step.rs
Comment (dj8yfo) [reply cargo-near-build/src/near/docker_build/subprocess_step.rs:87]: Second thiught, this doesn't need to be reverted to debug, as it's quite adequate verbosity for --teach-me, and RUST_LOG=info in container won't trigger this line, as it happens only outside of container. But i somehow got confused it would be triggered 😕

Comment (dj8yfo) [reply cargo-near-build/src/near/docker_build/subprocess_step.rs:87]: It probably can be excluded from RUST_LOG=info and included into --teach_me by adjusting near_teach_me=debug but it's probably too much fuss around single debug statement

Comment (FroVolod) [reply cargo-near-build/src/near/docker_build/subprocess_step.rs:87]: I won't change it

File: cargo-near/src/main.rs
Comment (FroVolod) [reply cargo-near/src/main.rs:19]: Done

File: cargo-near-build/src/env_keys.rs
Comment (FroVolod) [reply cargo-near-build/src/env_keys.rs]: Done

File: cargo-near/src/main.rs
Comment (FroVolod) [reply cargo-near/src/main.rs]: Done

**Review by dj8yfo — APPROVED** [review]

**Pull request merged**
